contrib/mesos/pkg/executor/service/service.go

```diff
@@ -173,6 +173,19 @@ func (s *KubeletExecutorServer) runKubelet(
 		kcfg.DockerDaemonContainer = "" // don't move the docker daemon into a cgroup
 		kcfg.Hostname = kcfg.HostnameOverride
 		kcfg.KubeClient = apiclient
+
+		// taken from KubeletServer#Run(*KubeletConfig)
+		clientConfig, err := s.CreateAPIServerClientConfig()
+		if err == nil {
+			kcfg.KubeClient, err = client.New(clientConfig)
+
+			// make a separate client for events
+			eventClientConfig := *clientConfig
+			eventClientConfig.QPS = s.EventRecordQPS
+			eventClientConfig.Burst = s.EventBurst
+			kcfg.EventClient, err = client.New(&eventClientConfig)
```

Comment:
upstream has different requirements. they allow kubelet to run without a
client. we don't.

On Wed, Dec 9, 2015 at 10:53 AM, Sergiusz Urbaniak <notifications@github.com
> wrote:

> In contrib/mesos/pkg/executor/service/service.go
> <https://github.com/kubernetes/kubernetes/pull/18434#discussion_r47107995>
> :
>
> > @@ -173,6 +173,19 @@ func (s *KubeletExecutorServer) runKubelet(
> >  		kcfg.DockerDaemonContainer = "" // don't move the docker daemon into a cgroup
> >  		kcfg.Hostname = kcfg.HostnameOverride
> >  		kcfg.KubeClient = apiclient
> > +
> > +		// taken from KubeletServer#Run(*KubeletConfig)
> > +		clientConfig, err := s.CreateAPIServerClientConfig()
> > +		if err == nil {
> > +			kcfg.KubeClient, err = client.New(clientConfig)
> > +
> > +			// make a separate client for events
> > +			eventClientConfig := *clientConfig
> > +			eventClientConfig.QPS = s.EventRecordQPS
> > +			eventClientConfig.Burst = s.EventBurst
> > +			kcfg.EventClient, err = client.New(&eventClientConfig)
>
> good catch, upstream shouldn't do it either.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/18434/files#r47107995>.
>